qiskit/circuit/library/bit_flip_oracle.py

```diff
@@ -0,0 +1,127 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Bit-flip Oracle object."""
+
+from qiskit.circuit import QuantumCircuit
+
+from qiskit.synthesis.boolean.boolean_expression import BooleanExpression
+
+
+class BitFlipOracle(QuantumCircuit):
+    r"""Bit-flip Oracle.
+
+    The Bit-flip Oracle object constructs circuits for any arbitrary
+    input logical expressions. A logical expression is composed of logical operators
+    `&` (`AND`), `|` (`OR`), `~` (`NOT`), and `^` (`XOR`).
+    as well as symbols for literals (variables).
+    For example, `'a & b'`, and `(v0 | ~v1) & (~v2 & v3)`
+    are both valid string representation of boolean logical expressions.
+
+    A bit-flip oracle for a boolean function `f(x)` performs the following
+    quantum operation:
+
+    .. math::
+
+            |x\rangle|y\rangle \mapsto |x\rangle|f(x)\oplus y\rangle
+
+    For convenience, this oracle, in addition to parsing arbitrary logical expressions,
+    also supports input strings in the `DIMACS CNF format
+    <https://web.archive.org/web/20190325181937/https://www.satcompetition.org/2009/format-benchmarks2009.html>`__,
+    which is the standard format for specifying SATisfiability (SAT) problem instances in
+    `Conjunctive Normal Form (CNF) <https://en.wikipedia.org/wiki/Conjunctive_normal_form>`__,
+    which is a conjunction of one or more clauses, where a clause is a disjunction of one
+    or more literals. See :meth:`qiskit.circuit.library.bit_flip_oracle.BitFlipOracle.from_dimacs_file`.
+
+    From 16 variables on, possible performance issues should be expected when using the
+    default synthesizer.
+    """
+
+    def __init__(
+        self,
+        expression: str,
+        var_order: list = None,
+    ) -> None:
+        """Creates a BitFlipOracle object
+
+        Args:
+            expression: A Python-like boolean expression.
+            var_order(list): A list with the order in which variables will be created.
+               (default: by appearance)
```

Comment:
Some minor comments on docstrings (which might be off in the original class already, not that you added this 🙂)
*  if `None` is supported we can annotate it with `OtherType | None` (this might require `from __future__ import annotations`)
* you could also add the type of the list elements -- I'm not sure but this might be `list[str]`?
* if the type is in the signature it can be skipped in the docstring
* we typically skip the initial sentence in the initializer
```suggestion
        var_order: list[str] | None = None,
    ) -> None:
        """
        Args:
            expression: A Python-like boolean expression.
            var_order: A list with the order in which variables will be created.
               (default: by appearance)
```

Reply:
Done in [d94f490](https://github.com/Qiskit/qiskit/pull/13769/commits/d94f490eebf36759ec2f645516c72f0fdad746e2)